Framework/API/inc/MantidAPI/MatrixWorkspace.h

```diff
@@ -602,9 +602,9 @@ class MANTID_API_DLL MatrixWorkspace : public IMDWorkspace,
 
   /// Flag indicating whether the m_isCommonBinsFlag has been set. False by
   /// default
-  mutable bool m_isCommonBinsFlagSet{false};
+  mutable std::atomic<bool> m_isCommonBinsFlagSet{false};
   /// Flag indicating whether the data has common bins. False by default
-  mutable bool m_isCommonBinsFlag{false};
+  mutable std::atomic<bool> m_isCommonBinsFlag{false};
```

Comment:
Why does `m_isCommonBinsFlag` also have to be atomic? `MatrixWorkspace::invalidateCommonBinsFlag()` doesn't modify its value.

Reply:
Good point. I had flipped both of them without looking at the implementation.